File: Ed_Cf_Round_134/D_Maximum_AND.py
import math,sys;input=sys.stdin.readline;S=lambda:input().rstrip();I=lambda:int(S());M=lambda:map(int,S().split());L=lambda:list(M());mod1=1000000007;mod2=998244353
for _ in range(I()):
    n=I();a=L();b=L();s=list(range(n));l=[[s,s]];ans=0
    for j in range(30,-1,-1):
        t=(1<<j);p=[];c=0
        for s1,s2 in l:
            s10=[];s11=[];s20=[];s21=[]
            for i in s1:
                if (a[i]&t):s11.append(i)
                else:s10.append(i)
            for i in s2:
                if (b[i]&t):s21.append(i)
                else:s20.append(i)
            if len(s10)==len(s21):
                c+=1;p.append([s10,s21]);p.append([s11,s20])
        if c==len(l):l=p;ans+=t
    print(ans)

File: Ed_Cf_Round_134/D_Maximum_AND.java
import java.util.*;
public class D_Maximum_AND{
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int tt=sc.nextInt();
        while(tt-->0){
            int n=sc.nextInt();
            ArrayList<Integer> s=new ArrayList<>();
            for(int i=0;i<n;i++)s.add(i);
            ArrayList<ArrayList<Integer>> l=new ArrayList<>();
            l.add(s);
            l.add(s);
            long ans=0;
            ArrayList<Integer> a=new ArrayList<>();
            ArrayList<Integer> b=new ArrayList<>();
            for(int i=0;i<n;i++)a.add(sc.nextInt());
            for(int i=0;i<n;i++)b.add(sc.nextInt());
            for(int j=30;j>=0;j--){
                int t=(1<<j);
                int c=0;
                ArrayList<ArrayList<Integer>> p=new ArrayList<>();
                for(int k=0;k<l.size()-1;k+=2){
                    ArrayList<Integer> s1=l.get(k);
                    ArrayList<Integer> s2=l.get(k+1);
                    ArrayList<Integer> s10=new ArrayList<>();
                    ArrayList<Integer> s11=new ArrayList<>();
                    ArrayList<Integer> s20=new ArrayList<>();
                    ArrayList<Integer> s21=new ArrayList<>();
                    for(int z=0;z<s1.size();z++){
                        if((t&(a.get(s1.get(z))))>0)s11.add(s1.get(z));
                        else s10.add(s1.get(z));
                    }
                    for(int z=0;z<s2.size();z++){
                        if((t&(b.get(s2.get(z))))>0)s21.add(s2.get(z));
                        else s20.add(s2.get(z));
                    }
                    if(s10.size()==s21.size()){
                        p.add(s10);p.add(s21);p.add(s11);p.add(s20);c+=2;
                    }
                }
                if(c==l.size()){
                    ans+=t;l=p;
                }
            }
            System.out.println(ans);
        }
        sc.close();
    }
}